proposals/0000-ios-scenedelegate-and-resizing.md

```diff
@@ -0,0 +1,173 @@
+
+#### Migration: RCTDevLoadingView
+
+The does not account for updating the overlay `UIWindow` constraints so it does not get resized along with the main window. The proposed fix is to update the constraints based on KVO observation of the main window’s frame.
+
+![](assets/0000-RCTDevLoadingView-width.png)
+![](assets/0000-RCTDevLoadingView-width-clipped.png)
+
+#### Migrating Dimensions: internal state adjustment & naming convention
+
+Dimensions currently are exported as a constant. Such design does not fit the concept of resizable windows. To accommodate this, `RCTDeviceInfo` (iOS native module) should update its internal state variable upon changes to the frame so as to make `getConstants` return a value that is up-to-date at the time of invocation.
+
+`Dimensions` (JS API) contains a `getConstants` method that wraps the native `getConstants` method and caches the result internally. The caching would now be obsolete (since the value may change in time), therefore the underlying native method should always be invoked. Moreover, the naming of the JS `getConstants` method in face of resizable windows may be misleading, since from now on the dimensions are not constant. Therefore, I propose a gradual adoption of a new method of same functionality, `getInfo()`, along with the deprecation of `getConstants()`, on the JS API side. A rough draft of the JS API changes would be:  
+
+![](assets/0000-nativedeviceinfo-api.png)
+
+## Drawbacks
+
+### Migration from `AppDelegate` to `SceneDelegate`
+
+- In React Native source and RNTester: will be covered by the PR following this RFC  
+- In user RN applications using `UIApplicationSupportsMultipleScenes=NO`: adjusting Info.plist and the entrypoint according to the upgrade helper diff  
+- In user RN applications using `UIApplicationSupportsMultipleScenes=YES`:  
+  - Ones not using native code:  
+    - adjusting `Info.plist` and the entrypoint according to the upgrade helper diff  
+    - ensuring that consumed libraries work well in multi-instance setups   
+  - Ones using native code:  
+    - the above  
+    - migrating from `AppDelegate` lifecycle methods to `UIScene` lifecycle methods  
+    - inspecting usage of singletons & static fields to ensure they logically fit the multi-instance reality  
+- In RN libraries:  
+  - migrating from `AppDelegate` lifecycle methods to `UIScene` lifecycle methods  
+  - inspecting usage of singletons & static fields to ensure they logically fit the multi-instance reality
```

Comment:
These are not really drawbacks... I'd love to see this section reworded as an action plan, where each item is a task (`- [ ] Task 1`) so that we can track the implementation.

The only drawback of this RFC is hat users will have to migrate and we have to handle an ecosystem migration.